packages/react/src/components/OverflowMenuV2/OverFlowMenuV2-test.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright IBM Corp. 2016, 2024
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { act, fireEvent, render, screen } from '@testing-library/react';
+
+import { FeatureFlags } from '../FeatureFlags';
+import { MenuItem } from '../Menu';
+import { OverflowMenuV2 } from './';
+import React from 'react';
+import { action } from '@storybook/addon-actions';
+
+// Mocking console.warn to track deprecation warning
+jest.spyOn(console, 'warn').mockImplementation(() => {});
+
+describe('<OverflowMenuV2 />', () => {
+  let consoleWarnSpy;
+
+  beforeEach(() => {
+    consoleWarnSpy = jest.spyOn(console, 'warn').mockImplementation(() => {});
+    jest.clearAllMocks();
+  });
+
+  afterEach(() => {
+    consoleWarnSpy.mockRestore();
+  });
+
+  it('logs the deprecation warning when rendering OverflowMenuV2', () => {
+    const onClick = action('onClick (MenuItem)');
+
+    render(
+      <OverflowMenuV2>
+        <MenuItem label="Stop app" onClick={onClick} />
+      </OverflowMenuV2>
+    );
+
+    // Check if the deprecation warning is logged
+    expect(consoleWarnSpy).toHaveBeenCalledWith(
+      expect.stringContaining(
+        '`<OverflowMenuV2>` is deprecated and will be removed in the next major version.'
+      )
+    );
+  });
+
+  it('renders with FeatureFlags and passes the correct flag', async () => {
```

Comment:
Could we change test names for consistency? and try something like below
"should render correctly with feature flag enabled"

Reply:
Updated test names accordingly